xbmc/cores/VideoPlayer/VideoPlayer.cpp

```diff
@@ -3132,8 +3144,52 @@ int CVideoPlayer::GetSubtitle()
   return m_SelectionStreams.IndexOf(STREAM_SUBTITLE, *this);
 }
 
+void CVideoPlayer::UpdateStreamInfos()
+{
+  CSingleLock lock(m_player_status_lock);
+  int streamId;
+  std::string retVal;
+
+  // video
+  streamId = GetVideoStream();
+
+  if (streamId > 0 && streamId < GetVideoStreamCount())
+  {
+    SelectionStream& s = m_SelectionStreams.Get(STREAM_VIDEO, streamId);
+    s.bitrate = m_VideoPlayerVideo->GetVideoBitrate();
+
+    if (m_pDemuxer)
```

Comment:
I wasn't sure if this was needed for streams with resolution changes (e.g. live TV adverts). Does this not change here? Or is stream closed/opened?